### DISCUSSION TITLE
[ Implemented ]  Easier Constraint Identification Through Highlighting

### DISCUSSION THREAD

**Discussion by TheMakerTales** [hlorus/CAD_Sketcher#210] [Feature Requests]
Currently, we have a great constraint viability control system. However once you hide a few constraints of the same type it gets very hard to identify which is the hidden constraint interacting with an entity.

So currently it looks like this:
<img width="1148" alt="Capture" src="https://user-images.githubusercontent.com/103122109/176430334-c19aa703-2119-4545-ade6-c99e44760306.PNG">

I propose we have something that looks a little like the image below. This way you can see all the constraints that are connected to the currently selected entities.
<img width="1148" alt="Capture copy" src="https://user-images.githubusercontent.com/103122109/176430583-aa3c3609-cfc3-4fb7-a56b-f9e33371e386.png">

**Comment by TheMakerTales** [comment]
There could be an argument made that instead of highlighting them in what can be a HUGE list of constraints that yet another panel is created in the N menu called "Entity Constraints" and this panel is populated with the constraints of the selected entities.

**Reply by Intuos5** [reply]
But, with multiple panels and lists, the user is more prone to accidentally look at the wrong panel and get the wrong impression. Instead, I propose to add a separator, so that entity constraints are moved to the top of the list above the separator. With nothing selected, it would work as it does now.

**Comment by BaoNT-GH** [comment]
I actually had another thought when implementing the constraints' visibility toggling, something more to consider as "smart constraint visibility" that uses the same fundamental you mentioned in this feature discussion. That is, constraints could be hidden by default, when entities are active (by hovered or selected), it shows all the related constraints to those entities.

**Reply by TheMakerTales** [reply]
Sounds nice!
My only concern would be that I've hidden them cus they were in the way so if I was about to turn your show of selection on and off plus have them viewable in a panel in some way be it a separate "Entity Constraints" panel or the monster one I think that might make the best of both worlds.

**Reply by Intuos5** [reply]
This may also create a bit of a performance hit if the addon needs to compute which ones are and which ones aren't visible.
